app/src/pages/DataSync/AdministrationList.js

```diff
@@ -58,6 +60,58 @@ const FormSelection = ({ navigation, route }) => {
     filterData(search, data);
   }, [search, data]);
 
+  async function fetchData(administration, form, pageNumber = 1, allData = []) {
+    try {
+      const response = await api.get(
+        `/datapoint-list?page=${pageNumber}&administration=${administration}&form=${form}`,
+      );
+      const data = response.data.data;
+
+      const updatedData = [...allData, ...data];
+
+      if (data.hasMorePages) {
+        return fetchData(administration, form, pageNumber + 1, updatedData);
+      } else {
+        return updatedData;
+      }
+    } catch (error) {
+      throw error;
```

Comment:
Please show the error with it's error code like this: https://app.asana.com/0/1204676393419426/1204615985394950/f